api-gateway/src/main/resources/static/scripts/vet-details/vet-details.controller.js

```diff
@@ -214,12 +216,80 @@ angular.module('vetDetails')
+
+            // Send a POST request to add the new education
+            $http.post("api/gateway/vets/" + $stateParams.vetId + "/educations", education)
+                .then(function (resp) {
+                    console.log(resp.data);
+                    self.education = resp.data;
+                    self.addEducationFormVisible = false;
+                    alert('Your education was successfully added!');
+
+                    $http.get('api/gateway/vets/' + $stateParams.vetId + '/educations').then(function (resp) {
+                        console.log(resp.data);
+                        self.educations = resp.data;
+                    });
+                    // clear the form with .reset()
+
+                    document.getElementById("educationForm").reset();
+              
+                    // Refresh the education data
+                    $http.get('api/gateway/vets/' + $stateParams.vetId + '/educations').then(function (resp) {
+                        console.log(resp.data);
+                        self.educations = resp.data;
+                    });
+                }, function (error) {
+                    alert('Error adding education: ' + error.data.message);
+                });
         };
+
+        self.submitEducationForm = function (education) {
+            education.vetId = $stateParams.vetId;
+            education.degree = document.getElementById("degree").value;
+            education.schoolName = document.getElementById("schoolName").value;
+            education.fieldOfStudy = document.getElementById("fieldOfStudy").value;
+            education.startDate = document.getElementById("startDate").value;
+            education.endDate = document.getElementById("endDate").value;
+
+            // Send a POST request to add the new education
+            $http.post("api/gateway/vets/" + $stateParams.vetId + "/educations", education)
+                .then(function (resp) {
+                    console.log(resp.data);
+                    self.education = resp.data;
+                    self.addEducationFormVisible = false;
+                    alert('Your education was successfully added!');
+
+                    $http.get('api/gateway/vets/' + $stateParams.vetId + '/educations').then(function (resp) {
+                        console.log(resp.data);
+                        self.educations = resp.data;
+                    });
+                    // clear the form with .reset()
+
+                    document.getElementById("educationForm").reset();
+
+                    // Refresh the education data
+                    $http.get('api/gateway/vets/' + $stateParams.vetId + '/educations').then(function (resp) {
+                        console.log(resp.data);
+                        self.educations = resp.data;
+                    });
+                }, function (error) {
+                    alert('Error adding education: ' + error.data.message);
+                });
+        };
```

Comment:
I think you accidentally duplicated your method here.

Reply:
ty miss mila its been resolved